workbench-core/accounts/src/utilities/hostingAccountLifecycleService.ts

```diff
@@ -651,12 +668,12 @@ export default class HostingAccountLifecycleService {
   }): Promise<void> {
     const { statusHandlerArn, artifactBucketArn, mainAcctEncryptionArnList } = arns;
 
-    // Update main account default event bus to accept hosting account state change events
-    await this.updateBusPermissions(statusHandlerArn, awsAccountId);
-
     // Add account to artifactBucket's bucket policy
     await this.updateArtifactsBucketPolicy(artifactBucketArn, awsAccountId);
 
+    // Update main account default event bus to accept hosting account state change events
+    await this.updateBusPermissions(statusHandlerArn, awsAccountId);
```

Comment:
i think this is ok for now since we're short on time, but relying on the order seems brittle since its not clear that call order for the functions is critical to things working correctly. would it be possible to know whether the account is valid by the time we get here so we know we're safe? i'm guessing no but thought i'd ask just in case.

Reply:
I agree that this is brittle. There isn't a clear cut way to figure out if the accountId actually belongs to a real aws account. AWS doesn't have an API for that. It only comes up as an error through using other APIs that rely on real account ids. 

Maybe adding a comment about the order will be helpful?